Print the path to the code with `TORCH_LOGS=output_code`

lezcano opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #99038

This is quite useful to play around with the code once it's been generated.

cc @soumith @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

lezcano committed:
Print the path to the code with `TORCH_LOGS=output_code`

This is quite useful to play around with the code once it's been generated.

[ghstack-poisoned]

peterbell10 commented on torch/_inductor/graph.py:
```diff
@@ -673,7 +673,7 @@ def compile_to_module(self):
             setattr(mod, name, value)
 
-        log.debug("Output code written to: %s", mod.__file__)
+        output_code_log.debug("Output code written to: %s", mod.__file__)
```
Can we have it so it's still possible to get the filename ouput without dumping the code. It's pretty annoying to have to scroll past pages of console output to get to the filename.

lezcano replied:
As it's going to require a bit more effort to do that, let's do it in a different PR. Leaving it as `info` now so that it's already future-proof.

lezcano replied:
Actually, this can be done setting `TORCH_LOGS="inductor"` in this PR. In master, it can be done setting `+inductor`. While writing this PR I've realised that the logging system has some issues as well, see https://github.com/pytorch/pytorch/issues/99144

lezcano committed:
Update on "Print the path to the code with `TORCH_LOGS=output_code`"


This is quite useful to play around with the code once it's been generated.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

lezcano committed:
Update on "Print the path to the code with `TORCH_LOGS=output_code`"


This is quite useful to play around with the code once it's been generated.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

lezcano commented on test/dynamo/test_logging.py:
```diff
@@ -104,7 +104,7 @@ def test_dynamo_error(self, records):
 
     test_aot = within_range_record_test(2, 6, aot=logging.INFO)
-    test_inductor_debug = within_range_record_test(3, 15, inductor=logging.DEBUG)
+    test_inductor_debug = within_range_record_test(2, 15, inductor=logging.DEBUG)
```
I believe this happened because of https://github.com/pytorch/pytorch/issues/99144.

mlazos commented on torch/_inductor/graph.py:
```diff
@@ -673,7 +673,7 @@ def compile_to_module(self):
             setattr(mod, name, value)
 
-        log.debug("Output code written to: %s", mod.__file__)
+        output_code_log.info("Output code written to: %s", mod.__file__)
```
Hmm I'm thinking maybe we should just log it to both the `output_code` log and normal inductor log here. It will print twice in cases where `TORCH_LOGS=+inductor, output_code` but I think that's acceptable.

lezcano committed:
Update on "Print the path to the code with `TORCH_LOGS=output_code`"


This is quite useful to play around with the code once it's been generated.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

lezcano commented:
@mlazos ptal

lezcano commented:
@pytorchbot merge -r

lezcano committed:
Update on "Print the path to the code with `TORCH_LOGS=output_code`"


This is quite useful to play around with the code once it's been generated.

cc soumith voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx peterbell10 desertfire

[ghstack-poisoned]

lezcano commented:
@pytorchbot merge
